src/builtins_str.rs

```diff
@@ -787,6 +789,104 @@ fn builtin_char_is_whitespace(
     ))
 }
 
+fn builtin_char_int(
+    environment: &mut Environment,
+    args: &mut dyn Iterator<Item = Expression>,
+) -> Result<Expression, LispError> {
+    let next_arg = param_eval(environment, args, "char->int")?;
+    let next_arg_d = next_arg.get();
+    params_done(args, "char->int")?;
+    let to_int = |s: &str| -> Result<Expression, LispError> {
+        let mut count = 0;
+        for _s in <str as UnicodeSegmentation>::graphemes(s, true) {
+            count += 1;
+        }
+        match count {
+            0 => Ok(Expression::alloc_data(ExpEnum::Int(0))),
+            1 => {
+                let mut int_val: u32 = 0;
+                let mut overflow = false;
+                for c in s.chars() {
+                    let (add, overflowed) = int_val.overflowing_add(c as u32);
+                    if overflowed {
+                        overflow = true;
+                        break;
+                    }
+                    int_val += add;
+                }
+                if overflow {
+                    Err(LispError::new("error, overflow occurred adding unicode scalar values that compose provided grapheme interpreted as unsigned 32 byte integers: {:?}."))
+                } else {
+                    Ok(Expression::alloc_data(ExpEnum::Int(int_val as i64)))
+                }
+            }
+            _ => Err(LispError::new(
+                "function takes one grapheme, multiple were provided.",
```

Comment:
In general (applies to all the errors you are raising) why did you not include the function name?  I know we talked about this once and yes the backtrace should probably take you to the call but including the name can just make things clearer (especially to not have to check your column in a long line of forms).  We can talk about this tonight though (maybe I am contradicting myself and we can run some tests but I think we will want the function names).